compiler/resolution/lowerIterators.cpp

```diff
@@ -2146,6 +2195,10 @@ static void removeUncalledIterators()
     if (! call->isPrimitive(PRIM_YIELD))
       continue;
 
+    // This is a backup check to ensure PRIM_INT_ERROR is never executed.
+    if (maybeCalled(call->parentSymbol, 4))
```

Comment:
I'm a little concerned about the heuristic nature of this.  If there's a function that's really never called but maybeCalled() returns true for it, it would be a compile time error instead of correctly running program.  Maybe it won't happen in practice though?